packages/api/src/getUserFromContext.ts

```diff
@@ -0,0 +1,45 @@
+import type { APIGatewayProxyEvent, Context } from 'aws-lambda'
+import fetch from 'node-fetch'
+
+export type SupportedAuthTypes = 'auth0' | 'netlify'
+
+type NetlifyClientContext = Context & {
+  clientContext: {
+    user?: any
+  }
+}
+
+// TODO: Define what an Auth0 user looks like.
+const REDWOOD_AUTH_TYPE_HEADER = 'x-redwood-auth-type'
+
+export const getUserFromContext = async ({
+  event,
+  context,
+}: {
+  event: APIGatewayProxyEvent
+  context: NetlifyClientContext
+}) => {
+  const type = event?.headers[REDWOOD_AUTH_TYPE_HEADER] as SupportedAuthTypes
+  switch (type) {
```

Comment:
If using an `enum` above for `SupportAuthTypes`, could use something like https://github.com/UselessPickles/ts-enum-util here